packages/core/src/components/tabs/tabs.tsx

```diff
@@ -330,20 +331,14 @@ export class Tabs extends AbstractPureComponent<TabsProps, TabsState> {
             return undefined;
         }
 
-        const tabTitleId = generateTabTitleId(this.props.id, id);
-        const tabPanelId = generateTabPanelId(this.props.id, id);
-
         return (
-            <div
-                aria-labelledby={tabTitleId}
-                aria-hidden={id !== this.state.selectedTabId}
-                className={classNames(Classes.TAB_PANEL, className, panelClassName)}
-                id={tabPanelId}
+            <TabPanel
+                {...tab.props}
                 key={id}
-                role="tabpanel"
-            >
-                {Utils.isFunction(panel) ? panel({ tabTitleId, tabPanelId }) : panel}
-            </div>
+                className={classNames(className, panelClassName)}
+                isHidden={id !== this.state.selectedTabId}
```

Comment:
since any consumer would need to set this same logic up I wonder if we should have the prop be `selectedTabId` instead so that users can't do weird things like have multiple tab panels showing

Reply:
Eh the logic may not always be the same. Take, for example, this case where `isHidden` is always `false` because there's always only 1 `TabPanel` being rendered:

```tsx
import * as React from "react";
import { Tab, Tabs, TabPanel, type TabId } from "@blueprintjs/core";

const TABS_PARENT_ID = React.useId();
const [selectedTabId, setSelectedTabId] = React.useState<TabId>("Home");

<Tabs
    id={TABS_PARENT_ID}
    onChange={setSelectedTabId}
    selectedTabId={selectedTabId}
>
    <Tab id="Home" title="Home" />
    <Tab id="Files" title="Files" />
</Tabs>
...
<TabPanel
    id={selectedTabId}
    isHidden={false}
    parentId={TABS_PARENT_ID}
    panel={<p>The current panel id is: "{selectedTabId}"</p>}
/>

Reply:
yea I think I'm leaning towards `selectedTabId` because then we control more of the implementation - I'm worried about trying to use the same `TabPanel` for multiple tabs causing issues down the line somehow